src/cargo/ops/cargo_test.rs

```diff
@@ -194,6 +194,10 @@ fn run_doc_tests(
                 CompileKind::Host => {}
                 CompileKind::Target(target) => {
                     if target.short_name() != compilation.host {
+
```

Comment:
Thanks for the contribution.

We need a test for this warning to make sure it doesn't emit excessive warnings. 

https://github.com/rust-lang/cargo/issues/12118#issuecomment-1541860834

Reply:
currently it will show every time a doctest is passed over (i.e skipped)

else we are gonna need to use something like, 

`pub static ref WARNONCE: Mutex<bool> = Mutex::new(false);`

(btw i really like lazy_static crate but i kinda admire  package count of a self hosted compiler)

Reply:
This part of cargo is single-threaded, so perhaps a simple variable here serves the purpose.

https://github.com/rust-lang/cargo/blob/94d274d6920aaabf44ee56a786dd0ac6f99759a0/src/cargo/ops/cargo_test.rs#L181

Reply:
done, pushed it with without locally building.... man as someone working with very slow computer gotta say i just 💖 your CI/CD set up.

Reply:
done, all tests have passed